tricycle_controller/CMakeLists.txt

```diff
@@ -47,7 +47,9 @@ target_include_directories(tricycle_controller PUBLIC
 )
 target_link_libraries(tricycle_controller PUBLIC tricycle_controller_parameters)
 ament_target_dependencies(tricycle_controller PUBLIC ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_compile_definitions(tricycle_controller PRIVATE _USE_MATH_DEFINES)
+# Causes the visibility macros to use dllexport rather than dllimport,
+# which is appropriate when building the dll but not consuming it.
+target_compile_definitions(tricycle_controller PRIVATE "TRICYCLE_CONTROLLER_BUILDING_DLL" "_USE_MATH_DEFINES")
```

Comment:
what is the difference of `TRICYCLE_CONTROLLER_BUILDING_DLL` to `TRICYCLE_CONTROLLER__VISIBILITY_BUILDING_DLL` as it was used with the other files?

Reply:
Basically `TRICYCLE_CONTROLLER_BUILDING_DLL` is used in the `visibility_control.h` header used for this library, see https://github.com/ros-controls/ros2_controllers/blob/9ab848a5a0f07a68b1cbf53c407fc162d0896a27/tricycle_controller/include/tricycle_controller/visibility_control.h#L33 , while `TRICYCLE_CONTROLLER__VISIBILITY_BUILDING_DLL` is is not referenced anywhere, if I am not wrong. More in general, indeed across the repo there are some packages that use macros that contain `__VISIBILITY` in the macro name, while other packages use macros without `__VISIBILITY`. However, in this PR I did not changed any existing macro name.

Reply:
OK. If there is a more general solution like you mentioned with CMake, maybe you could fix that in a future PR? We don't have to follow bad practices from "other ROS packages" ;)

Reply:
Cool! If you export all symbols by default on *nix, my favorite solution is to have the same on Windows, using `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` and avoiding to use any visibility_control header. I can open an issue for this to explain that if you like, is this repo ok or if you prefer something more general for the whole ros2_control ?

Reply:
For reference, some related discussions at the ros2-general level from the past:
* https://github.com/ament/ament_cmake/issues/201#issuecomment-542808202
* https://github.com/ament/ament_cmake/issues/164

Reply:
Any chance to do away with those visibility macros is very very welcome!

Reply:
@bmagyar @christophfroehlich I tried to recap the current situation at https://github.com/ros-controls/ros2_controllers/issues/1053 (sorry for the wall of text). Feel free to comment there, thanks!